test/e2e/csv_e2e_test.go

```diff
@@ -4433,88 +4427,83 @@ var csvAnyChecker = buildCSVConditionChecker(operatorsv1alpha1.CSVPhasePending,
-	Eventually(func() (bool, error) {
-		fetchedCSV, err = c.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	Eventually(func() bool {
+		fetched, err = c.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 		if err != nil {
-			return false, err
+			ctx.Ctx().Logf("error getting csv %s: %v", name, err)
+			return false
 		}
-		return checker(fetchedCSV), nil
+		ctx.Ctx().Logf("%s - %s (%s): %s", name, fetched.Status.Phase, fetched.Status.Reason, fetched.Status.Message)
+		return checker(fetched)
 	}).Should(BeTrue())
 
-	if err != nil {
-		ctx.Ctx().Logf("CSV %s/%s never got correct status: %#v", namespace, name, fetchedCSV.Status)
-	}
-	return fetchedCSV, err
+	// Keep signature, Eventually() will error exit on timeout, this only happens on success
+	return fetched, nil
 }
 
+// same function as fetchCSV(), but namespace and name reversed!
 func awaitCSV(c versioned.Interface, namespace, name string, checker csvConditionChecker) (*operatorsv1alpha1.ClusterServiceVersion, error) {
-	var fetched *operatorsv1alpha1.ClusterServiceVersion
-	var err error
-
-	Eventually(func() (bool, error) {
-		fetched, err = c.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				return false, nil
-			}
-			return false, err
-		}
-		ctx.Ctx().Logf("%s - %s (%s): %s", name, fetched.Status.Phase, fetched.Status.Reason, fetched.Status.Message)
-		return checker(fetched), nil
-	}).Should(BeTrue())
-
-	if err != nil {
-		ctx.Ctx().Logf("never got correct status: %#v", fetched.Status)
-	}
-	return fetched, err
+	return fetchCSV(c, name, namespace, checker)
 }
 
 func waitForDeployment(namespace string, c operatorclient.ClientInterface, name string) error {
-	return wait.Poll(pollInterval, pollDuration, func() (bool, error) {
+
+	Eventually(func() bool {
 		_, err := c.GetDeployment(namespace, name)
 		if err != nil {
-			if apierrors.IsNotFound(err) {
-				return false, nil
-			}
-			return false, err
+			ctx.Ctx().Logf("error getting deployment %s: %v", name, err)
+			return false
 		}
-		return true, nil
-	})
+		return true
+	}).WithTimeout(pollDuration).WithPolling(pollInterval).Should(BeTrue())
+
+	// Keep signature, Eventually() will error exit on timeout, this only happens on success
+	return nil
 }
 
 func waitForDeploymentToDelete(namespace string, c operatorclient.ClientInterface, name string) error {
-	var err error
-
-	Eventually(func() (bool, error) {
-		ctx.Ctx().Logf("waiting for deployment %s to delete", name)
+	ctx.Ctx().Logf("waiting for deployment %s to delete", name)
+	Eventually(func() bool {
 		_, err := c.GetDeployment(namespace, name)
-		if apierrors.IsNotFound(err) {
-			ctx.Ctx().Logf("deleted %s", name)
-			return true, nil
+		if apierrors.IsNotFound(err) || apierrors.IsGone(err) {
+			ctx.Ctx().Logf("deployment %s deleted", name)
```

Comment:
please ensure that namespaced resources are referred to with namespaced identifiers (namespace/name)

Reply:
Makes sense; this is a simple rewording of the prior message.